db/database.go

```diff
@@ -85,18 +112,76 @@ func (db *Database) ClearAllOffers() error {
 	defer iter.Release()
 
 	for iter.Valid() {
-		// key is the offer ID
+		offerID := iter.Key()
+		err := db.offerTable.Del(offerID)
+		if err != nil {
+			return err
+		}
+		iter.Next()
+	}
+
+	return nil
+}
+
+// PutSwap puts the given swap in the database.
+// If a swap with the same ID is already in the database, it overwrites it.
+func (db *Database) PutSwap(s *swap.Info) error {
+	val, err := json.Marshal(s)
+	if err != nil {
+		return err
+	}
+
+	key := s.ID
+	return db.swapTable.Put(key[:], val)
+}
+
+// HasSwap returns whether the db contains a swap with the given ID.
+func (db *Database) HasSwap(id types.Hash) (bool, error) {
+	return db.swapTable.Has(id[:])
+}
+
+// GetSwap returns a swap with the given ID, if it exists.
```

Comment:
Can we document if it returns `nil` or an error if it doesn't exist? Maybe add a unit test for the expected behavior too. (I looked at the tests to find the answer, but it wasn't in there.)

Reply:
it errors if it doesn't, will add a test